lib/model/message_list.dart

```diff
@@ -613,6 +619,15 @@ class MessageListView with ChangeNotifier, _MessageSequence {
   Anchor get anchor => _anchor;
   Anchor _anchor;
 
+  @override bool get oneMessagePerBlock => switch (narrow) {
+    CombinedFeedNarrow()
+      || ChannelNarrow()
+      || TopicNarrow()
+      || DmNarrow() => false,
+    MentionsNarrow()
+      || StarredMessagesNarrow() => true,
```

Comment:
I'd like to have a test that would fail if this started being all true or all false. Maybe a quick test case for a channel narrow, and another for the mentions narrow, showing different results with the same two messages.